### ISSUE TITLE
ASYNC0002 false positive when method returns ValueTask

### ISSUE THREAD

**Issue by EugenVeracity** [edumserrano/roslyn-analyzers#9]
```csharp
using System;
using System.Threading.Tasks;

public class TestClass : IAsyncDisposable
{
    public ValueTask DisposeAsync() // ASYNC0002: Remove Async suffix from method name
    {
        return ValueTask.CompletedTask;
    }
}
```

The analyser incorrectly flags a method returning ValueTask as non-async and suggests removing the Async suffix, which in this case is impossible because IAsyncDisposable is a built-in .NET interface. This does not happen when using Task as a return type, only ValueTask.
